packages/clients/core/src/index.ts

```diff
@@ -1071,6 +1071,16 @@ class Connector implements IConnector {
       this._eventManager.trigger({ event: "transport_close", params: [] }),
     );
 
+    this._transport.on("error", (e: Event)  => 
+      this._eventManager.trigger({
+        id: 0,
+        jsonrpc: '',
+        error: {
+          message: "Websocket connection failed"
+        },
+      })
+    );
+
```

Comment:
This hack will pass a `JsonRpcError` object to eventManager and the error will be caught under the `call_request` listener.